feat: add `stats/incr/nanmpcorrdist`

MohannedAhmed67 opened the pull request:
…cr/nanmpcorrdist

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #[5602](https://github.com/stdlib-js/stdlib/issues/5602).

## Description

> What is the purpose of this pull request?

This pull request:

-   It adds module which is a wrapper around mpcorrdist for handling NaN values. It ignores any NaN values passes in the accumulator

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #[5602](https://github.com/stdlib-js/stdlib/issues/5602) 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

MohannedAhmed67 committed:
feat: add new package to the stats/incr/* namespace: @stdlib/stats/incr/nanmpcorrdist

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

MohannedAhmed67 committed:
feat: add new package to the stats/incr/* namespace: @stdlib/stats/incr/nanmpcorrdist

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MohannedAhmed67 committed:
feat: add new package to the stats/incr/* namespace: @stdlib/stats/incr/nanmpcorrdist

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/README.md:
```diff
@@ -0,0 +1,190 @@
+
+-->
+
+# incrmpcorrdist
```
```suggestion
# incrnanmpcorrdist
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/README.md:
````diff
@@ -0,0 +1,190 @@
+var incrnanmpcorrdist = require( '@stdlib/stats/incr/nanmpcorrdist' );
+```
+
+#### incrmpcorrdist( window\[, mx, my] )
````
```suggestion
#### incrnanmpcorrdist( window\[, mx, my] )
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/README.md:
```diff
@@ -0,0 +1,190 @@
+    else {
+        x = randu() * 100.0;
+    }
+    if ( randu() < 0.2 ) {
```
Remove L144-149. You have copy-pasted code twice.

MohannedAhmed67 replied:
This is one for X and one for Y to make NaN values random for both of them and be generated independently.

hrshya replied:
Yeah I didn't notice but still it can be done in a single if-else statement.

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/README.md:
```diff
@@ -0,0 +1,190 @@
+
+<section class="related">
+
+* * *
```
Remove L163-169. This is autogenerated for new packages.

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/README.md:
```diff
@@ -0,0 +1,190 @@
+
+[pearson-correlation]: https://en.wikipedia.org/wiki/Pearson_correlation_coefficient
+
+<!-- <related-links> -->
```
Remove L180-186. Same this is also autogenerated for new packages.

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/benchmark/benchmark.js:
```diff
@@ -0,0 +1,91 @@
+	var i;
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		f = incrnanmpcorrdist( (i%5)+1 );
```
```suggestion
		f = incrnanmpcorrdist( ( i%5 ) +1 );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/docs/types/index.d.ts:
```diff
@@ -0,0 +1,97 @@
+* @returns accumulator function
+*
+* @example
+* var accumulator = incrmpcorrdist( 3, -2.0, 10.0 );
```
```suggestion
* var accumulator = incrnanmpcorrdist( 3, -2.0, 10.0 );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/docs/types/index.d.ts:
```diff
@@ -0,0 +1,97 @@
+* @returns accumulator function
+*
+* @example
+* var accumulator = incrmpcorrdist( 3 );
```
```suggestion
* var accumulator = incrnanmpcorrdist( 3 );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/docs/types/test.ts:
```diff
@@ -0,0 +1,123 @@
+* limitations under the License.
+*/
+
+import incrmpcorrdist = require( './index' );
```
```suggestion
import incrnanmpcorrdist = require( './index' );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/docs/types/test.ts:
```diff
@@ -0,0 +1,123 @@
+
+// The function returns an accumulator function...
+{
+	incrmpcorrdist( 3 ); // $ExpectType accumulator
```
```suggestion
	incrnanmpcorrdist( 3 ); // $ExpectType accumulator
```

hrshya replied:
Same comment for this entire files.

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/examples/index.js:
```diff
@@ -0,0 +1,50 @@
+	else {
+		x = randu() * 100.0;
+	}
+	if ( randu() < 0.2 ) {
```
Same comment as earlier you have copy-pasted twice here.

hrshya reviewed (commented):
Hey @MohannedAhmed67, thanks for your work on this! I’ve added some comments on the PR.

MohannedAhmed67 committed:
feat: add new package to the stats/incr/* namespace: @stdlib/stats/incr/nanmpcorrdist

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MohannedAhmed67 committed:
feat: add new package to the stats/incr/* namespace: @stdlib/stats/incr/nanmpcorrdist

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

MohannedAhmed67 commented:
@hrshya Thank you so much for your valuable comments on my PR !!!
Is there anything else left to be done ?

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+		} else {
+			d = data.slice( i-W+1, i+1 );
+		}
+		d = d.filter( function isNotNaN(pair) {
```
```suggestion
		d = d.filter( function isNotNaN( pair ) {
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+			d = data.slice( i-W+1, i+1 );
+		}
+		d = d.filter( function isNotNaN(pair) {
+			return !isnan(pair[0]) && !isnan(pair[1]);
```
```suggestion
			return !isnan( pair[ 0 ] ) && !isnan( pair[ 1 ] );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+			k = j;
+			while ( arr.length < min(countNonNanPairs, W) && k >= 0 ) {
+				if ( !isnan(d[k][0]) && !isnan(d[k][1]) ) {
+					arr.push(d[k]);
```
```suggestion
					arr.push( d[ k ] );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+			arr = [];
+			k = j;
+			while ( arr.length < min(countNonNanPairs, W) && k >= 0 ) {
+				if ( !isnan(d[k][0]) && !isnan(d[k][1]) ) {
```
```suggestion
				if ( !isnan( d[ k ][ 0 ] ) && !isnan( d[ k ][ 1 ] ) ) {
```

hrshya replied:
```suggestion
				if ( !isnan( d[ k ][ 0 ] ) && !isnan( d[ k ][ 1 ] ) ) {
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+
+			arr = [];
+			k = j;
+			while ( arr.length < min(countNonNanPairs, W) && k >= 0 ) {
```
```suggestion
			while ( arr.length < min( countNonNanPairs, W ) && k >= 0 ) {
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+		acc = incrnanmpcorrdist( W, means[ 0 ], means[ 1 ] );
+		countNonNanPairs = 0;
+		for ( j = 0; j < M; j++ ) {
+			actual = acc( d[j][0], d[j][1] );
```
```suggestion
			actual = acc( d[ j ][ 0 ], d[ j ][ 1 ] );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+		countNonNanPairs = 0;
+		for ( j = 0; j < M; j++ ) {
+			actual = acc( d[j][0], d[j][1] );
+			countNonNanPairs += ( !isnan(d[j][0]) && !isnan(d[j][1]) ) ? 1 : 0;
```
```suggestion
			countNonNanPairs += ( !isnan(d[ j ][ 0 ] ) && !isnan( d[ j ][ 1 ] ) ) ? 1 : 0;
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+
+	// For each dataset, compute the actual and expected correlation distances...
+	function isNotNaN(pair) {
+		return !isnan(pair[0]) && !isnan(pair[1]);
```
```suggestion
		return !isnan( pair[ 0 ] ) && !isnan( pair[ 1 ] );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+	W = 10;
+
+	// For each dataset, compute the actual and expected correlation distances...
+	function isNotNaN(pair) {
```
```suggestion
	function isNotNaN( pair ) {
```

hrshya replied:
Similar changes in the entire file.

hrshya reviewed (commented):
Hey @MohannedAhmed67, Good work. Just need some minor updates.

MohannedAhmed67 committed:
feat: add new package to the stats/incr/* namespace: @stdlib/stats/incr/nanmpcorrdist

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MohannedAhmed67 commented:
I have resolved all mentioned and gone through the file to fill all the similar minor issues.
Please check again @hrshya
Thanks in advance.

MohannedAhmed67 commented:
@hrshya Hello, have you checked the PR this time ?
I need to finish this because I started working on a new one yesterday, and I don't want to open many PRs without approving the current ones.

MohannedAhmed67 commented:
@kgryte Can I get a review of this PR ?

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/README.md:
```diff
@@ -0,0 +1,175 @@
+
+// For each simulated datum, update the moving sample correlation distance...
+for ( i = 0; i < 100; i++ ) {
+    if ( randu() < 0.2 ) {
```
We don't need separate if-else statements for x and y each as the conditions are same so it can be done in a single if-else statement.

MohannedAhmed67 replied:
The problem is that I want x to be NaN sometimes and y to be a value and vice versa, but if I included them in the the same if-else, x and y will be NaN at the same time,  and I want to include all the cases of 
(x, y)
(Number, Number)
(NaN, NaN)
(Number, NaN)
(NaN, Number)

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/examples/index.js:
```diff
@@ -0,0 +1,50 @@
+// For each simulated datum, update the moving sample correlation distance...
+console.log( '\nx\ty\tCorrelation Distance\n' );
+for ( i = 0; i < 100; i++ ) {
+	if ( randu() < 0.2 ) {
```
Same comment as earlier.

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/README.md:
```diff
@@ -0,0 +1,175 @@
+<!-- <div class="equation" align="center" data-raw-text="d_{x,y} = 1 - r_{x,y} = 1 - \frac{\operatorname{cov_n(x,y)}}{\sigma_x \sigma_y}" data-equation="eq:pearson_distance">
+    <img src="https://cdn.jsdelivr.net/gh/stdlib-js/stdlib@49d8cabda84033d55d7b8069f19ee3dd8b8d1496/lib/node_modules/@stdlib/stats/incr/mpcorrdist/docs/img/equation_pearson_distance.svg" alt="Equation for the Pearson product-moment correlation distance.">
+    <br>
+</div> -->
```
You can remove this, as this will be automatically populated.

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/README.md:
```diff
@@ -0,0 +1,175 @@
+
+#### incrnanmpcorrdist( window\[, mx, my] )
+
+Returns an accumulator `function` which incrementally computes a moving [sample Pearson product-moment correlation distance][pearson-correlation] while ignoring `NaN` values. The `window` parameter defines the number of values over which to compute the moving [sample Pearson product-moment correlation distance][pearson-correlation].
```
```suggestion
Returns an accumulator `function` which incrementally computes a moving [sample Pearson product-moment correlation distance][pearson-correlation], ignoring `NaN` values. The `window` parameter defines the number of values over which to compute the moving [sample Pearson product-moment correlation distance][pearson-correlation].
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/package.json:
```diff
@@ -0,0 +1,64 @@
+{
+    "name": "@stdlib/stats/incr/nanmpcorrdist",
+    "version": "0.0.0",
+    "description": "Compute a moving sample Pearson product-moment correlation distance incrementally while ignoring `NaN` values.",
```
```suggestion
    "description": "Compute a moving sample Pearson product-moment correlation distance incrementally, ignoring `NaN` values.",
```

hrshya replied:
Update this here and throughout the package to keep the consistency across the package.

hrshya reviewed (commented):
Just some minor changes. Rest, the PR looks good!

MohannedAhmed67 committed:
Merge branch 'incrnanmpcorrdist' of https://github.com/MohannedAhmed67/stdlib into develop

MohannedAhmed67 committed:
fixes

MohannedAhmed67 commented:
@hrshya Can you please review it again after the latest edits

hrshya commented:
Great work! Time to wait for a maintainer to merge this PR :)

MohannedAhmed67 commented:
@hrshya Do I need to mention anyone ?

MohannedAhmed67 commented:
@rotu @hrshya @rreusser @stdlib-bot 
Can anyone review this to be merged as it's been around one month of waiting for me to merge this PR.

MohannedAhmed67 commented:
@hrshya Updates ?

hrshya commented:
> @hrshya Updates ?

Maintainers must be busy because of the amount of PR they are getting. So the only thing we can do is wait for maintainers to review it and meanwhile you can open new PR's.

MohannedAhmed67 commented:
@hrshya

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/lib/main.js:
```diff
@@ -0,0 +1,99 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Copyright year for new files should be 2025. This applies to all files in this package.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/docs/types/index.d.ts:
```diff
@@ -0,0 +1,97 @@
+/// <reference types="@stdlib/types"/>
+
+/**
+* With ignoring `NaN` values, if provided input values, the accumulator function returns an updated moving sample correlation distance ignoring . If not provided input values, the accumulator function returns the current moving sample correlation distance.
```
Sentence is garbled — "ignoring ." ends with a dangling period and no object. Should read something like "...returns an updated moving sample correlation distance. If not provided input values..."

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/docs/types/index.d.ts:
```diff
@@ -0,0 +1,97 @@
+* ## Notes
+*
+* -   The correlation distance is defined as one minus the Pearson product-moment correlation coefficient and, thus, resides on the interval [0,2].
+* -   If provided `NaN` or a value which, when used in computations, results in `NaN`, the accumulated value is `NaN` for all future invocations.
```
This note says the accumulated value is `NaN` for all future invocations — that's the opposite of what this package does. Since this package skips `NaN` values, this note needs to be updated accordingly.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/docs/types/index.d.ts:
```diff
@@ -0,0 +1,97 @@
+type accumulator = ( x?: number, y?: number ) => number | null;
+
+/**
+* Returns an accumulator function which incrementally computes a moving sample Pearson product-moment correlation distance ignoring ignoring `NaN` values.
```
Duplicate word.

```suggestion
* Returns an accumulator function which incrementally computes a moving sample Pearson product-moment correlation distance, ignoring `NaN` values.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/lib/main.js:
```diff
@@ -0,0 +1,99 @@
+/**
+* Returns an accumulator function which incrementally computes a moving sample Pearson product-moment correlation distance, ignoring `NaN` values.
+*
+* @param {PositiveInteger} W - Sliding window size for the moving calculation. Must be a positive integer.
+* @param {number} [x] - Initial mean value of the first dataset. Defaults to `NaN`.
+* @param {number} [y] - Initial mean value of the second dataset. Defaults to `NaN`.
+* @returns {Function} Accumulator function which computes the Pearson correlation distance.
```
JSDoc parameter names should be `mx` and `my` to match the README signature and TypeScript declarations. Descriptions should be terse (e.g., `@param {number} [mx] - known mean`) and should not state "Defaults to NaN" — these are known mean values, not defaults.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/README.md:
```diff
@@ -0,0 +1,170 @@
+r = accumulator( NaN, NaN ); // [(NaN, 3.0), (4.0, NaN), (NaN, NaN)]
+// returns 1.0
+
+r = accumulator( 5.0, 2.0 ); // [(4.0, NaN), (NaN, NaN), 5.0, 2.0]
```
Missing parentheses around the last pair in the window comment.

```suggestion
r = accumulator( 5.0, 2.0 ); // [(4.0, NaN), (NaN, NaN), (5.0, 2.0)]
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+	}
+});
+
+tape('the function properly ignores NaN values in x and y inputs', function test( t ) {
```
Missing space after `tape(`.

```suggestion
tape( 'the function properly ignores NaN values in x and y inputs', function test( t ) {
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/test/test.js:
```diff
@@ -0,0 +1,671 @@
+	acc = incrnanmpcorrdist( 3, 3.14, 0.0 );
+	result = acc();
+
+	t.ok(!isnan( result ), 'result should not be NaN');
```
Missing spaces inside parentheses.

```suggestion
	t.ok( !isnan( result ), 'result should not be NaN' );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmpcorrdist/benchmark/benchmark.js:
```diff
@@ -0,0 +1,91 @@
+	var i;
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		f = incrnanmpcorrdist( ( i%5 ) +1 );
```
Missing spaces around operators.

```suggestion
		f = incrnanmpcorrdist( ( i % 5 ) + 1 );
```

Planeshifter reviewed (changes requested):
Sorry we didn't get to your PR. If you are still interested to bring it over the finish line, I have left a bunch of feedback.

Planeshifter committed:
fix: address review feedback for stats/incr/nanmpcorrdist

Planeshifter commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into incrnanmpcorrdist

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Planeshifter committed:
fix: correct keywords and README examples in stats/incr/nanmpcorrdist

Planeshifter committed:
fix: use isnan instead of negated condition in examples and README

Planeshifter committed:
fix: add parentheses around ternary conditions in examples

Planeshifter committed:
fix: use 2-space indentation in package.json

Planeshifter reviewed (approved):
Should be ready after clean-up. Thanks @MohannedAhmed67 and @hrshya!
